### PR TITLE
CPU-Strided-Complex Fixes for real and imag ops

### PR DESCRIPTION
**Pull request by dylanbespalko**
In-tree changes to pytorch to support complex numbers are being submitted here.
Out-of-tree support for complex numbers is here: [pytorch-cpu-strided-complex extension](https://gitlab.com/pytorch-complex/pytorch-cpu-strided-complex)

- [x]  Replaced std:real(a) with a.real() in kernel level code.
- [x]  Fixed Vec256_base implementation of complex ops so that it works correctly on Non-AVX devices. 
- [x]  Clean up CopyKernel after #29612 is approved.
@zasdfgbnm is fixing this issue in #29612. This should be added first.

cc: @iotamudelta, @ezyang, @bddppq

### CODE REVIEW

File: aten/src/ATen/cpu/vec256/vec256_base.h [hunk 1]
```diff
@@ -179,39 +179,75 @@ struct Vec256 {
     }
     return ret;
   }
-  template <typename other_t = T,
-            typename std::enable_if<!std::is_floating_point<other_t>::value && !std::is_complex_t<other_t>::value, int>::type = 0>
+  template <typename other_t_abs = T,
+            typename std::enable_if<!std::is_floating_point<other_t_abs>::value && !std::is_complex_t<other_t_abs>::value, int>::type = 0>
```
Comment (zasdfgbnm) [inline aten/src/ATen/cpu/vec256/vec256_base.h]: Does `std` has `is_complex_t`?

**Review by ezyang — APPROVED** [review]

File: aten/src/ATen/cpu/vec256/vec256_base.h (hunk 1 above)
Comment (dylanbespalko) [reply aten/src/ATen/cpu/vec256/vec256_base.h]: Weird. I was using the implementation in c10/util/Half.h.
The namespace should be C10. I'll change it.

**Comment by dylanbespalko** [comment]
@ezyang,

Can you please hold off accepting this code. I will likely pull changes from @zasdfgbnm and make his requested changes.

**Comment by ezyang** [comment]
OK

**Comment by zasdfgbnm** [comment]
@dylanbespalko In case you are not aware of this: https://github.com/pytorch/pytorch/pull/29631

**Comment by ezyang** [comment]
I messed up and forgot to wait sorry XD. Gonna unland.

**Comment by dylanbespalko** [comment]
No, Problem. I had to travel for a couple days so I haven't kept up. I'm getting back on this today.

**Pull request closed without merging**